File: FSchool-BE/Domain/Entities/Club.cs
using Domain.Enums;

namespace Domain.Entities
{
    public class Club
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string Name { get; set; }

        [MaxLength(50)]
        public string Category { get; set; } // VD: Academic, Sports, Arts

        public int MembersCount { get; set; }

        public string ImageUrl { get; set; }

        public string Description { get; set; }

        public ClubStatus Status { get; set; } = ClubStatus.PendingApproval;

        public DateTime FoundedDate { get; set; } = DateTime.UtcNow;

        // Giáo viên cố vấn
        public int? AdvisorStaffId { get; set; }
        [ForeignKey("AdvisorStaffId")]
        public Staff AdvisorStaff { get; set; }

        // Navigation Properties
        public ICollection<StudentClub> StudentClubs { get; set; }

        public ICollection<Event> Events { get; set; }
    }
}

File: StudentHub-FE/lib/models/club.dart
class Club {
  final String id;
  final String name;
  final String category;
  final int members;
  final String? image;
  final String? description;
  final String? status;
  final String? foundedDate;

  Club({
    required this.id,
    required this.name,
    required this.category,
    required this.members,
    this.image,
    this.description,
    this.status,
    this.foundedDate,
  });

  factory Club.fromJson(Map<String, dynamic> json) {
    return Club(
      id: json['id']?.toString() ?? '',
      name: json['name']?.toString() ?? '',
      category: json['category']?.toString() ?? '',
      members: (json['members'] as num?)?.toInt() ?? 0,
      image: json['image']?.toString(),
      description: json['description']?.toString(),
      status: json['status']?.toString(),
      foundedDate: json['foundedDate']?.toString(),
    );
  }
}
